pkg/credentialprovider/gcp/metadata.go

```diff
@@ -115,12 +117,27 @@ func init() {
 // Returns true if it finds a local GCE VM.
 // Looks at a product file that is an undocumented API.
 func onGCEVM() bool {
-	data, err := ioutil.ReadFile(gceProductNameFile)
-	if err != nil {
-		klog.V(2).Infof("Error while reading product_name: %v", err)
-		return false
+	var name string
+
+	if runtime.GOOS == "windows" {
+		data, err := exec.Command("wmic", "computersystem", "get", "model").Output()
```

Comment:
/cc @pjh

Reply:
I checked with a couple team members and they approved this approach. There's a small chance the result will be different in a shielded VM (with UEFI firmware instead of BIOS) - I'll test it out and will let you know if there's a difference (but don't block merge on this).